core/scripts/cli/sponsor/viewLiquidationDetails.js

```diff
@@ -1,18 +1,23 @@
 const inquirer = require("inquirer");
 const getDefaultAccount = require("../wallet/getDefaultAccount");
 const { LiquidationStatesEnum } = require("../../../../common/Enums.js");
-const { unwrapToEth } = require("./currencyUtils");
+const { getIsWeth, unwrapToEth } = require("./currencyUtils");
 const { submitTransaction } = require("./transactionUtils");
 
 const viewLiquidationDetails = async (web3, artifacts, emp, liquidation, id) => {
+  const ExpandedERC20 = artifacts.require("ExpandedERC20");
+
   const sponsorAddress = await getDefaultAccount(web3);
   const display = `Liquidated at epoch time ${liquidation.liquidationTime} by ${liquidation.liquidator}`;
   const backChoice = "Back";
   const withdrawAction = "Withdraw";
   choices = [{ name: backChoice }];
-  if (liquidation.state === LiquidationStatesEnum.DISPUTE_SUCCEEDED) {
-    // TODO: Need to detect whether you can withdraw or not?
+  // Check if the sponsor can withdraw by seeing if `withdrawLiquidation` reverts.
+  try {
+    await emp.withdrawLiquidation.call(id, sponsorAddress);
     choices.push({ name: withdrawAction });
+  } catch (err) {
+    // Withdraw wouldn't work so it shouldn't be a valid option.
```

Comment:
So if an error is caught and `withdrawAction` is not added to the `choices` array, the user will just see a `backChoice` option correct?

Reply:
Yeah, we only present the option if it'd work.